### PR TITLE
Fix invalid call to stat when FALLBACK_CAPATHS is empty

### PR DESCRIPTION
**Pull request by cmarcelo**
Loop wasn't taking into account the constant being set but empty. In
that case tok was set to NULL and stat was called with it. Replace
'do-while' with a 'while' to also perform the check the first time.

### CODE REVIEW

**Comment by tmarcu** [comment]
+1

File: src/curl.c [hunk 1]
```diff
@@ -95,7 +95,7 @@ int swupd_curl_init(void)
 		i = 1;
 
 		tok = strtok_r(str, ":", &ctx);
-		do {
+	        while (tok) {
```
Comment (matthewrsj) [inline src/curl.c]: Please run `make compliant` to make the whitespace here consistent with the rest of the project.

Comment (cmarcelo) [reply src/curl.c]: Done.

File: src/curl.c [hunk 2]
```diff
@@ -95,7 +95,7 @@ int swupd_curl_init(void)
 		i = 1;
 
 		tok = strtok_r(str, ":", &ctx);
-		do {
+		while (tok) {
 			if (stat(tok, &st)) {
```
Comment (icarus-sparry) [inline src/curl.c]: Don't you just hate infinite loops?

Consider what happens if stat fails!

Please can we fix this code so it is idiomatic, i.e. replace 98 with

    for ( tok = strtok_r(str, ":", &ctx); tok ; tock = strtok_r(NULL, ":", &ctx) {

and remove 97 and 111

Comment (cmarcelo) [reply src/curl.c]: Thanks for catching this :-)  Tested with valid paths but without invalid ones.

Done.

**Pull request merged**